Log meaningful job names and use template based log messages

kjac opened the pull request:
### Prerequisites

- [x] I have added steps to test this contribution in the description below

If there's an existing issue for this PR then this fixes https://github.com/umbraco/Umbraco-CMS/issues/15302

### Description

As described in the linked issue, the new `RecurringBackgroundJobHostedServiceRunner` has a little flaw in its logging: It doesn't log the actual names of the jobs it's starting, but the generic type of the hosted service instead:

![image](https://github.com/umbraco/Umbraco-CMS/assets/7405322/5249afc4-3b3c-45ae-af1e-d6977ce6cdc5)

This PR makes the logging a little bit more verbose, in order to log both the creation and the start-up of each individual hosted service - this time using the job names:

![image](https://github.com/umbraco/Umbraco-CMS/assets/7405322/b231f443-7206-403d-8207-1f3a67e556a4)

While I was here, I also changed all logging to use template based log messages.

### Testing this PR

All recurring jobs (hosted services) should start and run as per usual. The log should display the names of the jobs rather than the generic type of the hosted service.

Nuklon commented on src/Umbraco.Infrastructure/BackgroundJobs/RecurringBackgroundJobHostedServiceRunner.cs:
```diff
@@ -33,45 +26,45 @@ public class RecurringBackgroundJobHostedServiceRunner : IHostedService
             {
-                _logger.LogInformation($"Starting background hosted service for {hostedService.GetType().Name}");
+
+                _logger.LogInformation("Creating background hosted service for {job}", jobName);
```
I think this is a bit too much spam, there's already a "Starting ...", and if it fails a "Failed to start ..." message, there are no other possibilities.

Migaroez replied:
Creating a job and starting it can have 2 different failure reasons.
With the current code flow, if you see a `failed to start` message it might be because of the factory failing or the created service not actually starting.

A way to resolve this and reduce messages being logged is to Log a message right after getting the jobName `Booting/Initializing/Hosting background job {Jobname}`
And wrapping the factory and the start in separate try catches with each having a distinct message template
For Completnes, you could leave the Creating/Starting messages but change their level to debug.

kjac commented:
Makes sense - thanks for chiming in, @Nuklon 😄 

I have changed the "Creating..." message to use log level "Debug", so it won't be visible unless one is really looking for it. And @Migaroez is spot on why we need both messages 😆 

![image](https://github.com/umbraco/Umbraco-CMS/assets/7405322/90a9989a-80d0-4f5d-a031-88258655b47a)

Also added the class, thanks for the suggestion 👍
